From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Haylen Chu <heylenay@4d2.org>
Date: Mon, 14 Oct 2024 07:38:11 +0000
Subject: dt-bindings: thermal: sophgo,cv1800-thermal: Add Sophgo CV1800
 thermal

Add devicetree binding documentation for thermal sensors integrated in
Sophgo CV1800 SoCs.

Signed-off-by: Haylen Chu <heylenay@4d2.org>
---
 Documentation/devicetree/bindings/thermal/sophgo,cv1800-thermal.yaml | 57 ++++++++++
 1 file changed, 57 insertions(+)

diff --git a/Documentation/devicetree/bindings/thermal/sophgo,cv1800-thermal.yaml b/Documentation/devicetree/bindings/thermal/sophgo,cv1800-thermal.yaml
new file mode 100644
index 000000000000..111111111111
--- /dev/null
+++ b/Documentation/devicetree/bindings/thermal/sophgo,cv1800-thermal.yaml
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/thermal/sophgo,cv1800-thermal.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Sophgo CV1800 on-SoC Thermal Sensor
+
+maintainers:
+  - Haylen Chu <heylenay@4d2.org>
+
+description: Sophgo CV1800 on-SoC thermal sensor
+
+$ref: thermal-sensor.yaml#
+
+properties:
+  compatible:
+    enum:
+      - sophgo,cv1800-thermal
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  sample-rate-hz:
+    minimum: 1
+    maximum: 1908
+    default: 1
+
+  '#thermal-sensor-cells':
+    const: 0
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - interrupts
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/sophgo,cv1800.h>
+    #include <dt-bindings/interrupt-controller/irq.h>
+    thermal-sensor@30e0000 {
+        compatible = "sophgo,cv1800-thermal";
+        reg = <0x30e0000 0x100>;
+        clocks = <&clk CLK_TEMPSEN>;
+        interrupts = <16 IRQ_TYPE_LEVEL_HIGH>;
+        #thermal-sensor-cells = <0>;
+    };
+...
-- 
Armbian

